client/core/core.go

```diff
@@ -2392,6 +2392,50 @@ func (c *Core) handleReconnect(host string) {
 		log.Errorf("unable to authorize DEX at %s: %v", host, err)
 		return
 	}
+
+	dc.marketMtx.RLock()
+	for mktKey, mkt := range dc.marketMap {
+		dc.booksMtx.Lock()
+		_, foundMkt := dc.books[mktKey]
+		dc.booksMtx.Unlock()
+		if !foundMkt {
+			continue
+		}
+		req, err := msgjson.NewRequest(dc.NextID(), msgjson.OrderBookRoute, &msgjson.OrderBookSubscription{
+			Base:  mkt.BaseID,
+			Quote: mkt.QuoteID,
+		})
+		if err != nil {
+			log.Errorf("error encoding 'orderbook' request: %v", err)
+			continue
+		}
+
+		errChan := make(chan error, 1)
+		result := new(msgjson.OrderBook)
+		err = dc.Request(req, func(msg *msgjson.Message) {
+			errChan <- msg.UnmarshalResult(result)
+		})
+		if err != nil {
+			log.Errorf("error re-subscribing to %s orderbook: %v", mkt, err)
+			continue
+		}
+		err = extractError(errChan, requestTimeout, msgjson.OrderBookRoute)
+		if err != nil {
+			log.Error(err)
+			continue
+		}
+
+		booky := newBookie(func() { c.unsub(dc, mktKey) })
+		err = booky.Sync(result)
+		if err != nil {
+			log.Error(err)
+			continue
+		}
+		dc.booksMtx.Lock()
+		dc.books[mktKey] = booky
```

Comment:
One issue is that the e.g. webserver will still be listening on the old `bookie`'s `BookFeed`. We need a way to either 1) Pass the old `bookie`'s `BookFeed`s to the new `bookie`, or 2) notify the caller that the old feed is dead, possibly by closing the channel. See `marketSyncer` too.

I actually think solution 1 is the best.